packages/devtools_app/lib/src/service_manager.dart

```diff
@@ -795,36 +821,50 @@ class ServiceExtensionManager {
       return;
     }
 
-    assert(value != null);
-    if (value is bool) {
-      // TODO(kenz): stop special casing http timeline logging once it can be
-      // called via `callServiceExtension`. See
-      // https://github.com/dart-lang/sdk/issues/43628.
-      if (name == extensions.httpEnableTimelineLogging.extension) {
-        await _service.forEachIsolate((isolate) async {
-          await _service.httpEnableTimelineLogging(isolate.id, value);
-        });
-      } else {
+    final callExtension = () async {
+      assert(value != null);
+      if (value is bool) {
+        // TODO(kenz): stop special casing http timeline logging once it can be
+        // called via `callServiceExtension`. See
+        // https://github.com/dart-lang/sdk/issues/43628.
+        if (name == extensions.httpEnableTimelineLogging.extension) {
+          await _service.forEachIsolate((isolate) async {
+            await _service.httpEnableTimelineLogging(isolate.id, value);
+          });
+        } else {
+          await _service.callServiceExtension(
+            name,
+            isolateId: _isolateManager.selectedIsolate.id,
+            args: {'enabled': value},
+          );
+        }
+      } else if (value is String) {
+        await _service.callServiceExtension(
+          name,
+          isolateId: _isolateManager.selectedIsolate.id,
+          args: {'value': value},
+        );
+      } else if (value is double) {
         await _service.callServiceExtension(
           name,
           isolateId: _isolateManager.selectedIsolate.id,
-          args: {'enabled': value},
+          // The param name for a numeric service extension will be the last part
+          // of the extension name (ext.flutter.extensionName => extensionName).
+          args: {name.substring(name.lastIndexOf('.') + 1): value},
         );
       }
-    } else if (value is String) {
-      await _service.callServiceExtension(
-        name,
-        isolateId: _isolateManager.selectedIsolate.id,
-        args: {'value': value},
-      );
-    } else if (value is double) {
-      await _service.callServiceExtension(
-        name,
-        isolateId: _isolateManager.selectedIsolate.id,
-        // The param name for a numeric service extension will be the last part
-        // of the extension name (ext.flutter.extensionName => extensionName).
-        args: {name.substring(name.lastIndexOf('.') + 1): value},
-      );
+    };
+
+    final Isolate isolate =
+        await _service.getIsolate(_isolateManager.selectedIsolate.id);
+    // Do not try to call Dart IO extensions for a paused isolate.
+    if (extensions.isDartIoExtension(name) &&
+        isolate.pauseEvent.kind.contains('Pause')) {
+      callbacksOnIsolateResume
```

Comment:
maybe i'm missing something but why can't we just make these extensions work for paused vm isolates?

Reply:
From @bkonyi in a conversation we had earlier today:
"Service extensions registered via dart:developer in general shouldn't be called while isolates are paused since they execute Dart code on the isolate"

Reply:
we don't have to deal with it now but how do we handle the edge case of service extensions that need to be set at or soon after the application starts?